app/javascript/plugins/petition/index.tsx

```diff
@@ -130,6 +130,7 @@ export class Petition extends Plugin<IPetitionPluginConfig> {
         this.events.emit('action:submitted_success', { petition: this })
       )
       .then(() => this.onCompleteTransition())
+      .then(() => this.events.emit('petition:sign_success'))
```

Comment:
Let's use the existing event instead of creating a new one : https://github.com/SumOfUs/Champaign/blob/development/app/javascript/plugins/petition/index.tsx#L128

We want it to emit `petition:complete` and `petition:[ref]:complete`, so we need to change that line to use `this.emit` (so that it will prefix the event name):
```js
this.emit('complete', this)
```

We only need to bind to `petition:complete` on the other side.

Reply:
Made use of existing event @eyko